### PR TITLE
Add docs to librustc/hir/check_attr.rs

### PR DESCRIPTION
**Pull request by bjorn3**
Also moved `check_attribute` up to ease reading.

### CODE REVIEW

**Comment by Mark-Simulacrum** [comment]
Looks like there are some tidy errors here:

```
tidy check (x86_64-unknown-linux-gnu)
[00:02:31] tidy error: /checkout/src/librustc/hir/check_attr.rs:39: trailing whitespace
[00:02:31] tidy error: /checkout/src/librustc/hir/check_attr.rs:56: trailing whitespace
[00:02:31] tidy error: /checkout/src/librustc/hir/check_attr.rs:65: trailing whitespace
[00:02:32] some tidy checks failed
```

**Comment by bjorn3** [comment]
fixed tidy errors

File: src/librustc/hir/check_attr.rs [hunk 1]
```diff
@@ -35,11 +35,26 @@ impl Target {
     }
 }
 
+/// Visitor which check if all attributes are valid.
+///
+/// Note: Doesn't check if attributes are existing.
```
Comment (michaelwoerister) [inline src/librustc/hir/check_attr.rs]: I'm not sure if this doc comment is useful. I would prefer a module-level comment saying something like:
```rust
//! This module implements some validity checks for attributes.
//! In particular it verifies that `#[inline]` and `#[repr]` attributes are 
//! attached to items that actually support them and if there are
//! conflicts between multiple such attributes attached to the same 
//! item.
```

**Review by michaelwoerister — COMMENTED** [review]
Thanks for the PR, @bjorn3! I've added a comment below.

**Comment by bjorn3** [comment]
Changed it

**Comment by michaelwoerister** [comment]
Thank you! There is some trailing whitespace on some lines that makes `make tidy` fail:
https://travis-ci.org/rust-lang/rust/jobs/240885005#L721

Looks good to me otherwise.

**Comment by bjorn3** [comment]
Removed trailing whitespace

**Comment by Mark-Simulacrum** [comment]
Actually, could you squash these commits down to one please? Thanks!

@bors r-

**Comment by bjorn3** [comment]
Squashed

**Comment by Mark-Simulacrum** [comment]
Thanks!

@bors r=michaelwoerister rollup

**Pull request merged**